netty/src/main/java/io/grpc/transport/netty/GrpcSslContexts.java

```diff
@@ -93,7 +111,37 @@ public static SslContextBuilder forServer(
    * an application requires particular settings it should override the options set here.
    */
   public static SslContextBuilder configure(SslContextBuilder builder) {
+    // The default client/server providers in Netty are the same, so it doesn't matter which
+    // method we call.
+    ApplicationProtocolConfig protocolConfig =
+            selectApplicationProtocolConfig(SslContext.defaultServerProvider());
     return builder.ciphers(Http2SecurityUtil.CIPHERS, SupportedCipherSuiteFilter.INSTANCE)
-        .applicationProtocolConfig(DEFAULT_APN);
+        .applicationProtocolConfig(protocolConfig);
+  }
+
+  /**
+   * Attempts to select the best {@link ApplicationProtocolConfig} for the given
+   * {@link SslProvider}.
+   */
+  public static ApplicationProtocolConfig selectApplicationProtocolConfig(SslProvider provider) {
+    switch (provider) {
+      case JDK: {
+        if (JettyTlsUtil.isJettyAlpnConfigured()) {
+          return ALPN;
+        }
+        if (JettyTlsUtil.isJettyNpnConfigured()) {
+          return NPN;
+        }
+        throw new IllegalArgumentException("Jetty ALPN/NPN has not been properly configured.");
+      }
+      case OPENSSL:
```

Comment:
To be clear, this should really have a TODO noting it doesn't work, because OpenSSL does not support FATAL_ALERT, on neither [client](https://github.com/netty/netty/blob/58ad1e310686044aa5daa3b070dfc512519ec81e/handler/src/main/java/io/netty/handler/ssl/OpenSslContext.java#L415) nor [server](https://github.com/netty/netty/blob/58ad1e310686044aa5daa3b070dfc512519ec81e/handler/src/main/java/io/netty/handler/ssl/OpenSslContext.java#L249).